### ISSUE TITLE
is there doco for preferNameDirectory?

### ISSUE THREAD

**Issue by SimonCropp** [dotnet/templating#865]
i cant see it here https://github.com/dotnet/templating/wiki/%22Runnable-Project%22-Templates

**Comment by mlorbetske** [comment]
It doesn't seem to be documented in the article linked from the wiki (though it is used in the examples) and the property is lacking docs in the schema.

I'm the meantime: the property is used to determine whether a directory with the name given for the project should be created for the content. If true, a directory will be created for the content if the user has not specified an output directory (meaning that the content will be created in the current directory) and the user has specified a name (otherwise it's inferred that the name and target directory are the current directory). If false, a directory is not created for the content, irrespective of the values of the output and name parameters. However, if the directory specified by the output parameter does not exist, it will be created (irrespective of the value of preferNameDirectory). Additionally, if the content should always be placed in its own directory, without regard to the name and output decision logic above), simply put the template content in a directory (or change the value of the "target" property in a "sources" element).

**Comment by mlorbetske** [comment]
Actually, it is called out in the introductory post on writing templates https://blogs.msdn.microsoft.com/dotnet/2017/04/02/how-to-create-your-own-templates-for-dotnet-new/

**Comment by SimonCropp** [comment]
i dont like to be difficult. but blog posts are not documentation. they get out of date too easily

**Comment by mlorbetske** [comment]
Called this out for the docs work in #390 as well. Closing this to track it there.

I don't take your comments as being difficult, we know we've got substantial issues around docs right now and @sayedihashimi has been working to get things in a better place. Comments like this are great as it helps to know what things people are interested in knowing about the most & what specifically is currently lacking.
